pkg/cache/clusterqueue_snapshot.go

```diff
@@ -149,3 +149,17 @@ func (c *ClusterQueueSnapshot) DominantResourceShareWithout(wlReq resources.Flav
 	}
 	return dominantResourceShare(c, without)
 }
+
+func (c *ClusterQueueSnapshot) FindTopologyAssignments(
+	tasRequestsByFlavor map[kueue.ResourceFlavorReference][]TASPodSetRequests) TASAssignmentsResult {
+	result := make(TASAssignmentsResult)
+	for tasFlavor, tasRequestsForFlavor := range tasRequestsByFlavor {
+		if tasFlavorCache := c.TASFlavors[tasFlavor]; tasFlavorCache != nil {
```

Comment:
Are we fine with ignoring TAS flavors that are not in cache?

Reply:
Definitely not, but `c` is ClusterQueueSnapshot and as all snapshots it is modified in only one goroutine.
We already checked that `tasFlavor` is in `c.TASFlavors` in `WorkloadsTopologyRequests`. 
We could repeat the check here and output an error, but then we need to think what to make with it :) .

Reply:
I agree it is not ideal, and I rely on the external context. Let me know if you would like this improved and if you have something specific in mind.

Reply:
Added a comment along the lines, and removed the safety check. If the assumption is incorrect the code will blow with panic rather than hide the issue and make invalid assinment. PTAL. Marking as resolved.